duckduckgo/iOS
# Alert user about abnormal app conditions

Task/Issue URL: https://app.asana.com/0/72649045549333/1204420573063618/f

**Description**:
Gracefully handle unrecoverable:
- compilation errors,
- insufficient disk space errors,
- keychain errors.

**Steps to test this PR**:
1. Force usage of embedded tds (disable the part where we download tds from remote)
1. Remove parts of the embedded tds.json so it is no longer parsable.

1. Simulate insufficient disk space by setting shouldPresentInsufficientDiskSpaceAlertAndCrash to true

1. Simulate keychain error by placing the following code anywhere in the app: EmailManager().forceSignOut()

**Copy Testing**:


**Theme Testing**:
